### PR TITLE
 [move] Upgrade to the newest state of Move step #1

### PR DESCRIPTION
**Pull request by wrwg**
### Description

This updates our repo to the latest state of the Move repo. There are multiple breaking changes we need to reconcile:

- The CLI API had been changed; package sub-command is gone. Lead to refactoring some of the aptos CLI code.
- The representation of Visibility has changed. Script is not longer a visiblity class, instead there is a boolean is_entry
- The notation `public(script) fun` has been replaced by `public entry fun`. Moving forward, we should reserve the notion of a 'script' for the actual ones, and the entrypoint functions are those which can be called without a script.
- All package names go from upper case to lower, as in `Vector` -> `vector` and `BCS` -> `bcs`
- Various modules have been moved from the move-stdlib into the nursery.

This PR is the first step to reconcile this. The dependencies have been updated and the Move modules from the
move-stdlib renamed. Also nursery has been put for now in the search path, as some of the modules we use are now there.

2nd step of refactoring is to rename all our own modules to use lower case, as well as creating a new `aptos-stdlib` which contains aptos specific std library module. (E.g. events and errors are now Aptos specific, for now they are in the nursery.)

### Test Plan

Existing test (hopefully they are complete, but I'm sure something is broken and not covered)

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2018)

### CODE REVIEW

File: aptos-move/af-cli/src/main.rs [hunk 1]
```diff
@@ -28,12 +28,12 @@ pub enum AfCommands {
 fn main() -> Result<()> {
     let error_descriptions: ErrorMapping = bcs::from_bytes(cached_framework_packages::error_map())?;
     let args = AfCli::parse();
-    match &args.cmd {
```
Comment (geekflyer) [inline aptos-move/af-cli/src/main.rs:31]: @gregnazario  do we actually still use this CLI? I thought that's all done via the `aptos` CLI now?

Comment (gregnazario) [reply aptos-move/af-cli/src/main.rs:31]: Yes, there are some features that aren't in the aptos CLI but we should put in

Comment (wrwg) [reply aptos-move/af-cli/src/main.rs:31]: Lets keep this until we are sure we covered all features in the aptos cli.

**Comment by wrwg** [comment]
After 3rd failure on CI (but success locally) added CICD:skip-sdk-integration-test

**Comment by areshand** [comment]
Look great, not seeing any issue from my side.

**Review by davidiw — APPROVED** [review]
looks good from the rust side...

I suspect some of the examples might be broken, but it is a good chance for the community to contribute

File: aptos-move/framework/aptos-framework/sources/Coin.move [hunk 2]
```diff
@@ -401,23 +401,23 @@ module AptosFramework::Coin {
...
+        deposit(signer::address_of(source), coins_minted);
         move_to(source, FakeMoneyCapabilities {
             mint_cap,
             burn_cap
         });
     }
 
     #[test(source = @0x1, destination = @0x2)]
-    public(script) fun end_to_end(
+    public entry fun end_to_end(
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/Coin.move:412]: What's the diff between public(script) and public entry?

**Comment by movekevin** [comment]
> looks good from the rust side...
> 
> I suspect some of the examples might be broken, but it is a good chance for the community to contribute

@wrwg Did you run through the examples in https://aptos.dev/tutorials/aptos-quickstarts in case any of them is broken?

File: aptos-move/framework/aptos-framework/sources/Coin.move (hunk 2 above)
Comment (wrwg) [reply aptos-move/framework/aptos-framework/sources/Coin.move:412]: None, just syntax change. The name "script" for a function is really idiotic, scripts are the thing which we send for execution.

**Comment by wrwg** [comment]
> > looks good from the rust side...
> > I suspect some of the examples might be broken, but it is a good chance for the community to contribute
> 
> @wrwg Did you run through the examples in https://aptos.dev/tutorials/aptos-quickstarts in case any of them is broken?

Maybe. But if there aren't tests then things might be broken. I land anyway and we fix after.

**Pull request merged**